keras_cv/datasets/waymo/transformer.py

```diff
@@ -0,0 +1,632 @@
+        + loc[..., tf.newaxis, :]
+    )
+
+
+def box_3d_vehicle_to_global(box_3d: tf.Tensor, sdc_pose: tf.Tensor) -> tf.Tensor:
+    """Transforms 3D boxes from vehicle to global frame.
+
+    Args:
+      box_3d: [..., N, 7] 3d boxes in vehicle frame.
+      sdc_pose: [..., 4, 4] the SDC pose.
+
+    Returns:
+      The boxes in global frame.
+    """
+    center = box_3d[..., 0:3]
+    dim = box_3d[..., 3:6]
+    heading = box_3d[..., 6]
+
+    new_center = point_vehicle_to_global(center, sdc_pose)
+    new_heading = (
+        heading + tf.atan2(sdc_pose[..., 1, 0], sdc_pose[..., 0, 0])[..., tf.newaxis]
+    )
+
+    return tf.concat([new_center, dim, new_heading[..., tf.newaxis]], axis=-1)
+
+
+def build_tensors_from_wod_frame(frame: dataset_pb2.Frame) -> Dict[str, tf.Tensor]:
+    """Builds tensors from a Waymo Open Dataset frame.
+
+    Args:
+      frame: a Waymo Open Dataset frame.
+
+    Returns:
+      Flat dictionary of tensors.
+    """
+
+    frame_id_bytes = "{}_{}".format(frame.context.name, frame.timestamp_micros).encode(
+        encoding="ascii"
+    )
+    frame_id = tf.bitcast(
+        tf.fingerprint(tf.expand_dims(frame_id_bytes, 0))[0], tf.int64
+    )
+
+    timestamp_micros = tf.constant(frame.timestamp_micros, dtype=tf.int64)
+    pose = tf.convert_to_tensor(
+        value=np.reshape(np.array(frame.pose.transform), [4, 4]), dtype_hint=tf.float32
+    )
+
+    point_tensors = _get_point(frame, _MAX_NUM_NON_TOP_LIDAR_POINTS)
+    point_label_tensors = _get_point_label(frame, point_tensors.point_xyz)
+
+    # Transforms lidar frames to global coordinates.
+    point_tensors.point_xyz = point_vehicle_to_global(point_tensors.point_xyz, pose)
+    point_label_tensors.label_box = box_3d_vehicle_to_global(
+        point_label_tensors.label_box, pose
+    )
+
+    # Constructs final results.
+    num_points = point_tensors.point_xyz.shape[0]
+    return {
+        "frame_id": frame_id,
+        "timestamp_offset": tf.constant(0.0, dtype=tf.float32),
+        "timestamp_micros": timestamp_micros,
+        "pose": pose,
+        "point_xyz": point_tensors.point_xyz,
+        "point_feature": point_tensors.point_feature,
+        "point_mask": tf.ones([num_points], dtype=tf.bool),
+        "point_range_image_row_col_sensor_id": point_tensors.point_range_image_row_col_sensor_id,
+        "label_box": point_label_tensors.label_box,
+        "label_box_id": point_label_tensors.label_box_id,
+        "label_box_meta": point_label_tensors.label_box_meta,
+        "label_box_class": point_label_tensors.label_box_class,
+        "label_box_density": point_label_tensors.label_box_density,
+        "label_box_detection_difficulty": point_label_tensors.label_box_detection_difficulty,
+        "label_box_mask": point_label_tensors.label_box_mask,
+        "label_point_class": point_label_tensors.label_point_class,
+        "label_point_nlz": point_tensors.label_point_nlz,
+    }
+
+
+def build_tensors_for_augmentation(
```

Comment:
some offline discussion -- docstring that this is to convert range image to point cloud, and point users for a way to use range image directly?
(Do we need a parse range image function in the future?)

Reply:
Done, added docstring to build_tensors_from_wod_frame. The range image parsing function is in frame_utils from waymo_open_dataset.utils.